### PR TITLE
aaron/Sync LocalStorage

### PR DESCRIPTION
**Pull request by aaimio**

### CODE REVIEW

File: src/root_files/app/localstorage-sync.html [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+        <meta charset='utf-8' />
+        <meta name='referrer' content='origin' />
+        <script>
+            window.addEventListener('message', function(message) {
+                var restricted_origins = [
+                    'https://deriv.app', 
+                    'https://staging.deriv.app'
+                ];
+
+                // TODO: Remove always true condition.
```
Comment (negarn) [inline src/root_files/app/localstorage-sync.html]: should we remove true now?

File: src/root_files/app/localstorage-sync.html [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+                if (true || restricted_origins.includes(message.origin)) {
```
Comment (negarn) [inline src/root_files/app/localstorage-sync.html]: why call it restricted_origins?

File: src/templates/_common/includes/deriv-iframe.jsx [hunk 3]
```diff
@@ -0,0 +1,11 @@
+import React from 'react';
+
+const DerivIFrame = () => (
+    <iframe
+        id='localstorage-sync'
+        src='https://aaron-binary.binary.sx/br_iframe-sync/localstorage-sync.html'
```
Comment (negarn) [inline src/templates/_common/includes/deriv-iframe.jsx]: can we also commit here the real url

File: src/root_files/app/localstorage-sync.html (hunk 2 above)
Comment (aaimio) [reply src/root_files/app/localstorage-sync.html]: Messages should only be processed if coming from origins contained within `restricted_origins`. If we don't check the origin, any website could `postMessage` us with an invalid `active_loginid` or `client.accounts`.

Comment (negarn) [reply src/root_files/app/localstorage-sync.html]: could we then instead call it allowed_origins? :D

**Pull request merged**